lib/elixir/lib/exception.ex

```diff
@@ -1345,17 +1345,25 @@ defmodule UndefinedFunctionError do
   defp hint(module, function, arity, _loaded?) do
     downcased_module = downcase_module_name(module)
 
-    candidate =
-      Enum.find(:code.all_available(), fn {name, _, _} ->
-        downcase_module_name(name) == downcased_module
+    candidates =
+      Enum.filter(:code.all_available(), fn {name, _, _} = candidate ->
```

Comment:
now that this is a filter, we need to limit the number of candidates

Reply:
Suggestion, move the for comprehension here. Something like:


    stripped_module = strip_elixir_prefix(module)

    for {name, _, _} <- candidate,
        downcase_module_name(name) == downcased_module or
             String.ends_with?(List.to_string(name), stripped_module)
        {:module, module} <- [load_module(candidate)],
        function_exported?(module, function, arity),
        do: module